src/main.js

```diff
@@ -113,28 +124,39 @@ class MiniGraphCard extends LitElement {
 
     if (conf.show.graph === 'bar') {
       const entities = conf.entities.length;
-      if ((conf.hours_to_show * conf.points_per_hour) * entities > MAX_BARS) {
+      if (conf.hours_to_show * conf.points_per_hour * entities > MAX_BARS) {
         conf.points_per_hour = MAX_BARS / (conf.hours_to_show * entities);
         // eslint-disable-next-line no-console
-        console.warn('mini-graph-card: Not enough space, adjusting points_per_hour to ', conf.points_per_hour);
+        console.warn(
+          'mini-graph-card: Not enough space, adjusting points_per_hour to ',
+          conf.points_per_hour,
+        );
       }
     }
     if (!this.Graph) {
-      this.Graph = conf.entities.map(() => (
-        new Graph(
+      this.Graph = conf.entities.map(
+        () => new Graph(
           500,
           conf.height,
           [conf.show.fill ? 0 : conf.line_width, conf.line_width],
           conf.hours_to_show,
           conf.points_per_hour,
-        )
-      ));
+        ),
+      );
     }
 
     this.config = conf;
   }
 
+  firstUpdated() {
```

Comment:
What do you think about replacing firstUpdated with connectedCallback & also add disconnectedCallback?

```js
  connectedCallback() {
    super.connectedCallback();
    if (this.config.update_interval) {
      this.updateOnInterval();
      this.interval = setInterval(this.updateOnInterval, this.config.update_interval * 1000);
    }
  }

  disconnectedCallback() {
    clearInterval(this.interval)
    super.disconnectedCallback();
  }
```

That way we could prevent updating the history while in the background, and also clear the interval properly.
And I guess we would also get the initial data faster